src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -33,11 +35,26 @@ protected ExternalWorkspace run() throws Exception {
         if (step.getUpstream() == null) {
             // this is the upstream job
 
-            DiskAllocationStrategy allocationStrategy = new MostUsableSpaceStrategy(step.getDiskPoolId(), step.getDescriptor().getDiskPools());
+            String diskPoolId = step.getDiskPoolId();
+            if (diskPoolId == null) {
+                throw new Exception("Disk Pool ID was not provided as step parameter");
+            }
+
+            List<DiskPool> diskPools = step.getDescriptor().getDiskPools();
+            DiskAllocationStrategy allocationStrategy = new MostUsableSpaceStrategy(diskPoolId, diskPools);
             Disk disk = allocationStrategy.allocateDisk();
 
-            String pathOnDisk = computePathOnDisk(disk.getPhysicalPathOnDisk());
-            ExternalWorkspace externalWorkspace = new ExternalWorkspace(disk.getDiskId(), pathOnDisk);
+            String physicalPathOnDisk = disk.getPhysicalPathOnDisk();
+            if (physicalPathOnDisk == null) {
+                throw new Exception("Physical path on disk was not provided in the Jenkins global config");
```

Comment:
With multiple disk pools, showing the `diskPoolId` in the error message will help with debug.

If we think about a fallback, then I think it would be the node's Remote root directory. If this is too complicated, then I would prefer no fallback, just an error.